File: Data Structures & Algorithms/minimum-window-with-characters/submission-2.cs
public class Solution {
    public string MinWindow(string s, string t) {
        if (t.Length > s.Length)
            return "";

        int[] tCount = new int[58];
        int[] sCount = new int[58];

        for (int i = 0; i < t.Length; i++) {
            tCount[t[i] - 'A']++;
        }

        for (int i = 0; i < s.Length; i++) {
            sCount[s[i] - 'A']++;
        }

        for (int i = 0; i < s.Length; i++) {
            if (sCount[s[i] - 'A'] < tCount[s[i] - 'A'])
                return "";
        }

        int l = 0, r = s.Length - 1;

        while (l < r && (sCount[s[l] - 'A'] - 1) >= tCount[s[l] - 'A']) {
            sCount[s[l] - 'A']--;
            l++;
        }

        while (l <= r && (sCount[s[r] - 'A'] - 1) >= tCount[s[r] - 'A']) {
            sCount[s[r] - 'A']--;
            r--;
        }

        string result = "";

        while (l <= r) {
            result += s[l];
            l++;
        }

        return result;
    }
}

File: Data Structures & Algorithms/anagram-groups/submission-2.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:

        anagrams = {}

        for s in strs:
            letters = [0] * 26
            
            for c in s:
                letters[ord(c) - ord('a')] += 1

            string_numbers = [str(n) for n in letters]
            strKey = ":".join(string_numbers)
            if strKey not in anagrams:
                anagrams[strKey] = [s]
            else:
                anagrams[strKey].append(s)

        result = []
        for key in anagrams:
            result.append(anagrams[key])

        return result
            
            